File: 프로그래머스/0/120852. 소인수분해/소인수분해.js
function solution(n) {
    const primeFactors = new Set();
    let i = 2;
    let currentN = n;
    
    while (i * i <= currentN) {
        if (currentN % i === 0) {
            primeFactors.add(i);
            currentN = Math.trunc(currentN / i); 
        } else {
            i++;
        }
    }
    
    if (currentN > 1) {
        primeFactors.add(currentN);
    }
    
    return [...primeFactors].sort((a, b) => a - b);
}

File: 프로그래머스/0/120852. 소인수분해/소인수분해.py
def solution(n):
    prime_factors = set()
    i = 2
    
    while i * i <= n:
        if n % i == 0:
            prime_factors.add(i)
            n //= i 
        else:
            i += 1
            
    if n > 1:
        prime_factors.add(n)
        
    return sorted(list(prime_factors))